pkg/controller/agent/pod.go

```diff
@@ -50,63 +63,199 @@ var (
+				// also need to do is to roll Elastic Agent Pods to pick up the update CA. To do be able to do that, we are
+				// adding Fleet Server associations (which includes Elasticsearch) to config hash attached to Elastic Agent
+				// Pods.
+				if err := commonassociation.WriteAssocsToConfigHash(params.Client, fs.GetAssociations(), configHash); err != nil {
+					return nil, err
+				}
+			}
+		}
 	}
 
-	for i, association := range params.Agent.GetAssociations() {
+	if esAssociation != nil {
+		builder = builder.
+			WithCommand([]string{"/usr/bin/env", "bash", "-c", trustCAScript(path.Join(certificatesDir(esAssociation), CAFileName))})
+	}
+
+	builder = builder.
+		WithVolumeLikes(vols...).
+		WithResources(defaultFleetResources).
+		// needed to pick up fleet-setup.yml correctly
+		WithEnv(corev1.EnvVar{Name: "CONFIG_PATH", Value: "/usr/share/elastic-agent"})
+
+	return builder, nil
+}
+
+func getVolumesFromAssociations(associations []commonv1.Association) []volume.VolumeLike {
+	vols := []volume.VolumeLike{}
+	for i, association := range associations {
 		if !association.AssociationConf().CAIsConfigured() {
-			continue
+			return nil
 		}
 		caSecretName := association.AssociationConf().GetCASecretName()
-		caVolume := volume.NewSecretVolumeWithMountPath(
+		vols = append(vols, volume.NewSecretVolumeWithMountPath(
 			caSecretName,
 			fmt.Sprintf("%s-certs-%d", association.AssociationType(), i),
 			certificatesDir(association),
-		)
-		vols = append(vols, caVolume)
+		))
+	}
+	return vols
+}
+
+func getAssociatedFleetServer(params Params) (commonv1.Associated, error) {
+	assoc := association.GetAssociationOfType(params.Agent.GetAssociations(), commonv1.FleetServerAssociationType)
+	if assoc == nil {
+		return nil, nil
 	}
 
-	volumes := make([]corev1.Volume, 0, len(vols))
-	volumeMounts := make([]corev1.VolumeMount, 0, len(vols))
-	var initContainers []corev1.Container
+	fsRef := assoc.AssociationRef()
+	fs := agentv1alpha1.Agent{}
 
-	for _, v := range vols {
-		volumes = append(volumes, v.Volume())
-		volumeMounts = append(volumeMounts, v.VolumeMount())
+	if err := association.FetchWithAssociations(
+		params.Context,
+		params.Client,
+		reconcile.Request{NamespacedName: fsRef.NamespacedName()},
+		&fs,
+	); err != nil {
+		return nil, err
 	}
 
-	builder := defaults.NewPodTemplateBuilder(podTemplate, ContainerName).
-		WithLabels(labels).
-		WithResources(defaultResources).
-		WithDockerImage(spec.Image, container.ImageRepository(container.AgentImage, spec.Version)).
-		WithArgs("-e", "-c", ConfigMountPath).
-		WithVolumes(volumes...).
-		WithVolumeMounts(volumeMounts...).
-		WithInitContainers(initContainers...).
-		WithInitContainerDefaults()
+	return &fs, nil
+}
 
-	return builder.PodTemplate
+func trustCAScript(caPath string) string {
+	return fmt.Sprintf(`#!/usr/bin/env bash
+set -e
+cp %s /etc/pki/ca-trust/source/anchors/
+update-ca-trust
+/usr/bin/tini -- /usr/local/bin/docker-entrypoint -e
```

Comment:
I'm fine with that considering it's short term only... (otherwise I'd expect we don't override the container entrypoint in case things change in future versions of agent, and rather rely on init containers.)

Reply:
Yes, it's supposed to be short term.